src/sagemaker/utils.py

```diff
@@ -1362,3 +1371,47 @@ def update_nested_dictionary_with_values_from_config(
     )
 
     return inferred_config_dict
+
+
+def stringify_object(obj: Any) -> str:
+    """Returns string representation of object, returning only non-None fields."""
+    non_none_atts = {key: value for key, value in obj.__dict__.items() if value is not None}
+    return f"{type(obj).__name__}: {str(non_none_atts)}"
+
+
+def volume_size_supported(instance_type: str) -> bool:
+    """Returns True if SageMaker allows volume_size to be used for the instance type.
+
+    Raises:
+        ValueError: If the instance type is improperly formatted.
+    """
+
+    try:
+
+        # local mode does not support volume size
+        if instance_type.startswith("local"):
+            return False
+
+        parts: List[str] = instance_type.split(".")
+
+        if len(parts) == 3 and parts[0] == "ml":
```

Comment:
nit: can use some regex magic to check for instancetype input - https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/image_uris.py#LL266C10-L266C10

Reply:
Discussed offline that the other utility has different specifications and would be large blast radius to merge utilities in this PR.